api/proto/teleport/devicetrust/v1/device.proto

```diff
@@ -93,7 +93,32 @@ message DeviceCredential {
   // Unique identifier of the credential, defined client-side.
   string id = 1;
   // Device public key marshaled as a PKIX, ASN.1 DER.
+  // For TPMs, this is the public application key.
   bytes public_key_der = 2;
+  // The degree to which the device credential is attested.
+  AttestationType attestation_type = 3;
+  // If the device is a TPM device, stores the serial number from the TPM
+  // endorsement certificate.
+  // The certificate is not stored for security reasons, but the serial can be
+  // used to retrieve information about a specific unit from the manufacturer
+  // at a later date.
+  string tpm_serial = 4;
+  // If the device is a TPM device, stores the public AK in PKIX, ASN.1 DER
+  // form.
+  bytes tpm_attestation_key_der = 5;
+}
+
+// AttestationType indicates the degree to which the device credential has
+// been attested.
+enum AttestationType {
```

Comment:
Maybe:

```suggestion
enum DeviceAttestationType {
```

or do you think that's too much?

Reply:
Nope - Agreed!

Reply:
00fe8f688f